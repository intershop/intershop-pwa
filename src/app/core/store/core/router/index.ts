// API to access ngrx core router state
export * from './router.selectors';
export * from './router.operators';
