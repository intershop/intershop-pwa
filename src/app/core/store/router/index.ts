export * from './router.actions';
export * from './router.selectors';
export { CoreState } from '../core.state';
