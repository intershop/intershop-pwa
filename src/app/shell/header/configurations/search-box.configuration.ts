export interface SearchBoxConfiguration {
  /**
   * id to support the multiple use on a page
   */
  id?: string;
  /**
   * text for search button on search box, icon is used if no text is provided
   */
  buttonText?: string;
  /**
   * placeholder text for search input field
   */
  placeholder?: string;
  /**
   * if autoSuggest is set to true auto suggestion is provided for search box, else no auto suggestion is provided
   */
  autoSuggest?: boolean;
  /**
   * configures the number of suggestions if auto suggestion is provided
   */
  maxAutoSuggests?: number;
  /**
   * configure search box icon
   */
  icon?: string;
  /**
   * show last search term as search box value
   */
  showLastSearchTerm?: boolean;
}
