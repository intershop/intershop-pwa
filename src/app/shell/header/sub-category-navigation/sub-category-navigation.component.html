<ul
  *ngIf="category.hasChildren()"
  class="category-level{{ subCategoriesDepth }}"
  [ngClass]="{ 'dropdown-menu': subCategoriesDepth === 1 }"
>
  <li
    *ngFor="let subcategory of category.children()"
    class="main-navigation-level{{ subCategoriesDepth }}-item"
    [ngClass]="{ open: isOpened(subcategory) }"
  >
    <a [routerLink]="subcategory | ishCategoryRoute" [ngStyle]="{ width: !subcategory.hasChildren() ? '100%' : '' }">{{
      subcategory.name
    }}</a>
    <a
      *ngIf="subcategory.hasChildren()"
      class="dropdown-toggle"
      [ngClass]="{ 'd-none': view === 'full' }"
      (click)="toggleOpen(subcategory)"
    >
      <fa-icon *ngIf="isOpened(subcategory); else closed" [icon]="['fas', 'minus']"></fa-icon>
      <ng-template #closed> <fa-icon [icon]="['fas', 'plus']"></fa-icon> </ng-template>
    </a>
    <ish-sub-category-navigation
      *ngIf="subCategoriesDepth < mainNavigationMaxSubCategoriesDepth"
      [category]="subcategory"
      [subCategoriesDepth]="subCategoriesDepth + 1"
    ></ish-sub-category-navigation>
  </li>
</ul>
