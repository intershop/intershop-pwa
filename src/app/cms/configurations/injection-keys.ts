import { InjectionToken, SimpleChange, Type } from '@angular/core';

import { ContentPageletView } from 'ish-core/models/content-view/content-views';

export interface CMSComponentInterface {
  pagelet: ContentPageletView;
  cmsDQNAttribute: string;
  ngOnChanges?(changes?: { pagelet: SimpleChange }): void;
}

export interface CMSComponentProvider {
  definitionQualifiedName: string;
  class: Type<CMSComponentInterface>;
}

export const CMS_COMPONENT = new InjectionToken<CMSComponentProvider>('InjectionToken');
