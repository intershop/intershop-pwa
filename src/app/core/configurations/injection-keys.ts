import { InjectionToken } from '@angular/core';

import { CookieConsentOptions } from 'ish-core/models/cookies/cookies.model';
import { ViewType } from 'ish-core/models/viewtype/viewtype.types';

import { environment } from '../../../environments/environment';

/**
 * Array of paths that always use mocked data
 */
export const API_MOCK_PATHS = new InjectionToken<string[]>('apiMockPaths', { factory: () => environment.apiMockPaths });

/**
 * If 'username' login name is used for registration, if 'email' the email is used as login name (default: 'email')
 */
export const USER_REGISTRATION_LOGIN_TYPE = new InjectionToken<string>('userRegistrationLoginType', {
  factory: () => 'email',
});

/**
 * The maximum subcategories level depth rendered in the main navigation
 */
export const MAIN_NAVIGATION_MAX_SUB_CATEGORIES_DEPTH = new InjectionToken<number>(
  'mainNavigationMaxSubCategoriesDepth',
  { factory: () => environment.mainNavigationMaxSubCategoriesDepth }
);

/**
 * global definition of the product listing page size
 */
export const PRODUCT_LISTING_ITEMS_PER_PAGE = new InjectionToken<number>('productListingItemsPerPage', {
  factory: () => environment.productListingItemsPerPage,
});

/**
 * default definition of the product listing view type
 */
export const DEFAULT_PRODUCT_LISTING_VIEW_TYPE = new InjectionToken<ViewType>('defaultProductListingViewType', {
  factory: () => environment.defaultProductListingViewType,
});

/**
 * the configured cookie consent options for the application
 */
export const COOKIE_CONSENT_OPTIONS = new InjectionToken<CookieConsentOptions>('cookieConsentOptions', {
  factory: () => environment.cookieConsentOptions,
});

/*
 * global definition of the Bootstrap grid system breakpoint widths
 */

export const MEDIUM_BREAKPOINT_WIDTH = new InjectionToken<number>('mediumBreakpointWidth', {
  factory: () => environment.mediumBreakpointWidth,
});

export const LARGE_BREAKPOINT_WIDTH = new InjectionToken<number>('largeBreakpointWidth', {
  factory: () => environment.largeBreakpointWidth,
});
