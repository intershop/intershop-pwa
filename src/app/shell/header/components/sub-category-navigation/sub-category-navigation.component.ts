import { ChangeDetectionStrategy, Component, Inject, Input } from '@angular/core';

import { MAIN_NAVIGATION_MAX_SUB_CATEGORIES_DEPTH } from 'ish-core/configurations/injection-keys';
import { CategoryView } from 'ish-core/models/category-view/category-view.model';
import { Category } from 'ish-core/models/category/category.model';

/**
 * The Sub Category Navigation Component displays second level category navigation.
 *
 * @example
 * <ish-sub-category-navigation
 *   [category]="category"
 *   [subCategoriesDepth]="1"
 * >
 * </ish-sub-category-navigation>
 */
@Component({
  selector: 'ish-sub-category-navigation',
  templateUrl: './sub-category-navigation.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class SubCategoryNavigationComponent {
  @Input() view = 'auto';
  @Input() category: CategoryView;
  @Input() subCategoriesDepth: number;

  openedCategories = [];

  constructor(@Inject(MAIN_NAVIGATION_MAX_SUB_CATEGORIES_DEPTH) public mainNavigationMaxSubCategoriesDepth: number) {}

  /**
   * Indicate if specific category is expanded.
   * @param category The category item.
   */
  isOpened(category: Category): boolean {
    return this.openedCategories.includes(category.uniqueId);
  }

  /**
   * Toggle category open state.
   * @param category The category item.
   */
  toggleOpen(category: Category) {
    const index = this.openedCategories.findIndex(id => id === category.uniqueId);
    index > -1 ? this.openedCategories.splice(index, 1) : this.openedCategories.push(category.uniqueId);
  }
}
