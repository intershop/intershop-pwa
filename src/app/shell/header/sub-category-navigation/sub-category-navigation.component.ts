import { ChangeDetectionStrategy, Component, Inject, Input, OnInit } from '@angular/core';
import { Observable } from 'rxjs';

import { MAIN_NAVIGATION_MAX_SUB_CATEGORIES_DEPTH } from 'ish-core/configurations/injection-keys';
import { ShoppingFacade } from 'ish-core/facades/shopping.facade';
import { NavigationCategory } from 'ish-core/models/navigation-category/navigation-category.model';

/**
 * The Sub Category Navigation Component displays second level category navigation.
 */
@Component({
  selector: 'ish-sub-category-navigation',
  templateUrl: './sub-category-navigation.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class SubCategoryNavigationComponent implements OnInit {
  @Input() view = 'auto';
  @Input() categoryUniqueId: string;
  @Input() subCategoriesDepth: number;

  openedCategories: string[] = [];

  navigationCategories$: Observable<NavigationCategory[]>;

  constructor(
    private shoppingFacade: ShoppingFacade,
    @Inject(MAIN_NAVIGATION_MAX_SUB_CATEGORIES_DEPTH) public mainNavigationMaxSubCategoriesDepth: number
  ) {}

  ngOnInit() {
    this.navigationCategories$ = this.shoppingFacade.navigationCategories$(this.categoryUniqueId);
  }

  /**
   * Indicate if specific category is expanded.
   */
  isOpened(uniqueId: string): boolean {
    return this.openedCategories.includes(uniqueId);
  }

  /**
   * Toggle category open state.
   */
  toggleOpen(uniqueId: string) {
    const index = this.openedCategories.findIndex(id => id === uniqueId);
    index > -1 ? this.openedCategories.splice(index, 1) : this.openedCategories.push(uniqueId);
  }
}
