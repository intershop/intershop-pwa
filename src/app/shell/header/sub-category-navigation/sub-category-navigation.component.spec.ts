import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { FaIconComponent } from '@fortawesome/angular-fontawesome';
import { MockComponent } from 'ng-mocks';

import { MAIN_NAVIGATION_MAX_SUB_CATEGORIES_DEPTH } from 'ish-core/configurations/injection-keys';
import { createCategoryView } from 'ish-core/models/category-view/category-view.model';
import { Category } from 'ish-core/models/category/category.model';
import { CategoryRoutePipe } from 'ish-core/routing/category/category-route.pipe';
import { categoryTree } from 'ish-core/utils/dev/test-data-utils';

import { SubCategoryNavigationComponent } from './sub-category-navigation.component';

describe('Sub Category Navigation Component', () => {
  let fixture: ComponentFixture<SubCategoryNavigationComponent>;
  let component: SubCategoryNavigationComponent;
  let element: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [CategoryRoutePipe, MockComponent(FaIconComponent), SubCategoryNavigationComponent],
      providers: [{ provide: MAIN_NAVIGATION_MAX_SUB_CATEGORIES_DEPTH, useValue: 2 }],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SubCategoryNavigationComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;

    const tree = categoryTree([
      { uniqueId: 'A', name: 'CAT_A', categoryPath: ['A'] },
      { uniqueId: 'A.1', name: 'CAT_A1', categoryPath: ['A', 'A.1'] },
      { uniqueId: 'A.2', name: 'CAT_A2', categoryPath: ['A', 'A.2'] },
      { uniqueId: 'A.1.a', name: 'CAT_A1a', categoryPath: ['A', 'A.1', 'A.1.a'] },
      { uniqueId: 'A.1.a.alpha', name: 'CAT_A1aAlpha', categoryPath: ['A', 'A.1', 'A.1.a', 'A.1.a.alpha'] },
    ] as Category[]);

    component.category = createCategoryView(tree, 'A');
    component.subCategoriesDepth = 1;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchSnapshot();
  });
});
