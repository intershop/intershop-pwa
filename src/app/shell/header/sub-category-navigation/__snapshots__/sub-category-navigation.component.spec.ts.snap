// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sub Category Navigation Component should be created 1`] = `
<ul class="category-level1 dropdown-menu">
  <li class="main-navigation-level1-item">
    <a ng-reflect-router-link="/category/A.1" href="/category/A.1">CAT_A1</a
    ><a class="dropdown-toggle"><fa-icon ng-reflect-icon-prop="fas,plus"></fa-icon></a
    ><ish-sub-category-navigation ng-reflect-sub-categories-depth="2"
      ><ul class="category-level2">
        <li class="main-navigation-level2-item">
          <a ng-reflect-router-link="/category/A.1.a" href="/category/A.1.a">CAT_A1a</a
          ><a class="dropdown-toggle"><fa-icon ng-reflect-icon-prop="fas,plus"></fa-icon></a>
        </li></ul
    ></ish-sub-category-navigation>
  </li>
  <li class="main-navigation-level1-item">
    <a style="width: 100%;" ng-reflect-router-link="/category/A.2" href="/category/A.2">CAT_A2</a
    ><ish-sub-category-navigation ng-reflect-sub-categories-depth="2"></ish-sub-category-navigation>
  </li>
</ul>
`;
